### PR TITLE
[Docs] [Remake] Add `app` command to `mesheryctl` reference

### LINKED ISSUE #3829: [Docs] Add `app` to mesheryctl documentation
#### Current State
`app` is missing on the Using mesheryctl docs page - https://docs.meshery.io/guides/mesheryctl/using-mesheryctl and on the https://docs.meshery.io/reference/mesheryctl reference.

---

#### Contributor [Guides](https://docs.meshery.io/project/contributing) and Resources
- 🛠 [Meshery Build & Release Strategy](https://docs.meshery.io/project/build-and-release)
- 📚 [Instructions for contributing to documentation](https://github.com/meshery/meshery/blob/master/CONTRIBUTING.md#documentation-contribution-flow)
   - Meshery documentation [site](https://docs.meshery.io/) and [source](https://github.com/meshery/meshery/tree/master/docs)
- 🎨 Wireframes and designs for Meshery UI in [Figma](https://www.figma.com/file/SMP3zxOjZztdOLtgN4dS2W/Meshery-UI)
- 🙋🏾🙋🏼 Questions: [Layer5 Discussion Forum](https://discuss.layer5.io) and [Layer5 Community Slack](http://slack.layer5.io)

### PR DESCRIPTION
**Pull request by alphaX86**
Signed-off-by: Aadhitya A <aadhitya864@gmail.com>

**Description**

This PR fixes #3829 
This PR is a remake of #3841 

**Notes for Reviewers**
Addl changes are also made, so go through them (eg: removal of extra `lang` field in all ref md files)

**[Signed commits](https://github.com/meshery/meshery/blob/master/CONTRIBUTING.md#signing-off-on-commits-developer-certificate-of-origin)**
- [x] Yes, I signed my commits.

### CODE REVIEW

**Comment by afzal442** [comment]
Nailed it! 🙄 so many files updated! GJ! 🚀  
Thanks so much!

File: docs/_data/mesheryctlcommands/cmds.yml
Comment (afzal442) [inline docs/_data/mesheryctlcommands/cmds.yml:526]:
```suggestion
          mesheryctl app view BookInfo (application-name|application-id)
          OR
          mesheryctl app view BookInfo
```
not sure! but it'll look good;

Comment (afzal442) [reply docs/_data/mesheryctlcommands/cmds.yml:526]: similarly in other areas!

**Review by afzal442 — CHANGES_REQUESTED** [review]

File: docs/_data/mesheryctlcommands/cmds.yml
Comment (navendu-pottekkat) [reply docs/_data/mesheryctlcommands/cmds.yml:526]: +1 to @afzal442 's point. Let's be consistent with other commands here.

**Comment by alphaX86** [comment]
@navendu-pottekkat @afzal442 Rog' 👍

**Comment by l5io** [comment]
🚀 Preview for commit 510aa4828778415b3e98fe8fe0fc68757785a189 at: https://614abb38745aee3ddd1a7cea--meshery-docs-preview.netlify.app

File: docs/pages/guides/mesheryctl/working-with-mesheryctl.md
Comment (navendu-pottekkat) [inline docs/pages/guides/mesheryctl/working-with-mesheryctl.md:14]:
```suggestion
- `mesheryctl app` - Service Mesh Application Management
```

File: docs/_data/mesheryctlcommands/cmds.yml
Comment (navendu-pottekkat) [inline docs/_data/mesheryctlcommands/cmds.yml:485]:
```suggestion
  description: Service Mesh application management
```

File: docs/_data/mesheryctlcommands/cmds.yml
Comment (navendu-pottekkat) [inline docs/_data/mesheryctlcommands/cmds.yml:502]:
```suggestion
      description: offboard application
```

File: docs/_data/mesheryctlcommands/cmds.yml
Comment (navendu-pottekkat) [inline docs/_data/mesheryctlcommands/cmds.yml:508]:
```suggestion
          description: Offboard application with an application file```

File: docs/_data/mesheryctlcommands/cmds.yml
Comment (navendu-pottekkat) [inline docs/_data/mesheryctlcommands/cmds.yml:513]:
```suggestion
      description: onboard application
```

File: docs/_data/mesheryctlcommands/cmds.yml
Comment (navendu-pottekkat) [inline docs/_data/mesheryctlcommands/cmds.yml:515]:
```suggestion
          mesheryctl app onboard [app name] [flags]
```

File: docs/_data/mesheryctlcommands/cmds.yml
Comment (navendu-pottekkat) [inline docs/_data/mesheryctlcommands/cmds.yml:504]:
```suggestion
          mesheryctl app offboard [app name] [flags]
```

File: docs/_data/mesheryctlcommands/cmds.yml
Comment (navendu-pottekkat) [inline docs/_data/mesheryctlcommands/cmds.yml:541]: Don't we have a separate examples section? Usage would be the skeleton of the command and not actual examples right?

**Review by navendu-pottekkat — CHANGES_REQUESTED** [review]

File: docs/_data/mesheryctlcommands/cmds.yml
Comment (alphaX86) [reply docs/_data/mesheryctlcommands/cmds.yml:541]: Examples section isn't made yet (for all cmds)

Comment (alphaX86) [reply docs/_data/mesheryctlcommands/cmds.yml:541]: There is an open issue made but no one has taken it yet

**Comment by l5io** [comment]
🚀 Preview for commit 6e8245842607e5dc0ae932ed136821a8af4608bd at: https://614acb63da0f945da1aaf007--meshery-docs-preview.netlify.app

File: docs/_data/mesheryctlcommands/cmds.yml
Comment (alphaX86) [reply docs/_data/mesheryctlcommands/cmds.yml:541]: @navendu-pottekkat FYI, #3978 

IDK who will take this issue but it'd be a good one to address it if anyone's willing to work on Jekyll

**Comment by l5io** [comment]
🚀 Preview for commit 26fc909730cc475e3547f56ee22cd800576d64ad at: https://614acc4b159f7366ea1050ba--meshery-docs-preview.netlify.app

File: docs/_data/mesheryctlcommands/cmds.yml
Comment (afzal442) [inline docs/_data/mesheryctlcommands/cmds.yml:526]: hey, you beat me off Rog! I suggested you let's make it either one of them; similarly in the examples section; e.g. https://614acc4b159f7366ea1050ba--meshery-docs-preview.netlify.app/reference/mesheryctl/apps#examples
https://614acc4b159f7366ea1050ba--meshery-docs-preview.netlify.app/reference/mesheryctl/apps/onboard#examples

File: docs/_data/mesheryctlcommands/cmds.yml
Comment (afzal442) [reply docs/_data/mesheryctlcommands/cmds.yml:541]: It could be either one of them; `mesheryctl app view BookInfo -o json` looks better.

File: docs/pages/reference/mesheryctl/subcommands/mesheryctl-app-view.md
Comment (afzal442) [inline docs/pages/reference/mesheryctl/subcommands/mesheryctl-app-view.md:27]:
```suggestion
  mesheryctl app view [application-name| application-id] | [flag]
```
let's remove everything and make it this way.

Comment (afzal442) [reply docs/pages/reference/mesheryctl/subcommands/mesheryctl-app-view.md:27]: everything else lgtm!

File: docs/_data/mesheryctlcommands/cmds.yml
Comment (alphaX86) [reply docs/_data/mesheryctlcommands/cmds.yml:526]: WAIT... It's still there? 🤔 I corrected them 😤 ok I'll check again

**Comment by l5io** [comment]
🚀 Preview for commit 46f406f7f6f2f3c1e9b6d60a4c5aa336644b313f at: https://614ae66320dd1280cb34cc3e--meshery-docs-preview.netlify.app

File: docs/_data/mesheryctlcommands/cmds.yml
Comment (alphaX86) [reply docs/_data/mesheryctlcommands/cmds.yml]: To do in Issue #3978

Comment (alphaX86) [reply docs/_data/mesheryctlcommands/cmds.yml]: For now, no examples. So I leave it for this PR

**Comment by l5io** [comment]
🚀 Preview for commit 0035706b2d7cc6957ca83b3fc0a1d3d661a425e3 at: https://614af2e4a307188714a60af3--meshery-docs-preview.netlify.app

**Comment by alphaX86** [comment]
@navendu-pottekkat if this PR is fine (I'll fix examples in other PR for all cmds and subcmds) could you review and merge (if possible) it so that it'd be a bit easy?

**Review by navendu-pottekkat — CHANGES_REQUESTED** [review]
Can you change occurrences of "Service Mesh Apps Management" with "Service Mesh Application Management"?

**Comment by navendu-pottekkat** [comment]
I believe I had mentioned this on the earlier review. Good to merge after that.

**Comment by alphaX86** [comment]
Done! 👍

**Comment by l5io** [comment]
🚀 Preview for commit 83023e3abf7af0a17852fec10d23f2e76630232b at: https://614afc07709e4c9c9b0bb1ce--meshery-docs-preview.netlify.app

**Comment by afzal442** [comment]
Thanks @alphaX86! let's brainstorm a bit more before we merge it. DM you on slack;

You have missed out on the comments I suggested!

**Comment by alphaX86** [comment]
> Thanks @alphaX86! let's brainstorm a bit more before we merge it. DM you on slack;
> 
> You have missed out on the comments I suggested!

Nah I never missed, job would be a huge though that's why I said

**Comment by navendu-pottekkat** [comment]
Application

Screenshot 2021-09-22 at 5 17 50 PM

**Comment by alphaX86** [comment]
> Application

Couldn't get you... A bit more clear?

**Comment by alphaX86** [comment]
> Application
> 
> Screenshot 2021-09-22 at 5 17 50 PM

OOOH I GET IT! 👍

**Comment by l5io** [comment]
🚀 Preview for commit 58f8a1fbb9e47dbdfa01de422112f51611199108 at: https://614b323b0dcdb0e9ede7eefe--meshery-docs-preview.netlify.app

**Comment by l5io** [comment]
🚀 Preview for commit fc8c6c43f7d0fab0af814fa77ca234241f031666 at: https://614b3373963c31c514284cea--meshery-docs-preview.netlify.app

**Comment by l5io** [comment]
🚀 Preview for commit cc2ff8543217b11b3f08e00f056bbfcc5adcbc64 at: https://614b3586745aeed7161a7bea--meshery-docs-preview.netlify.app

File: docs/pages/reference/mesheryctl/subcommands/mesheryctl-app-view.md
Comment (afzal442) [inline docs/pages/reference/mesheryctl/subcommands/mesheryctl-app-view.md:28]: you can remove it now as we are indicating here as `mesheryctl app view [application-name| application-id] [flag]` already, even it is in the `desc` section.

File: docs/_data/mesheryctlcommands/cmds.yml
Comment (afzal442) [inline docs/_data/mesheryctlcommands/cmds.yml:539]: let's make it in a square bracket;

Since you are interested to refactor each example in a different PR, let's ignore the working example for now 👍🏻 . I also made these changes without working examples 😄  earlier but due to misconception, we had to rework here ⛑️;

// @navendu-pottekkat

File: docs/_data/mesheryctlcommands/cmds.yml
Comment (afzal442) [inline docs/_data/mesheryctlcommands/cmds.yml:548]:
```suggestion
          description: display the content of an application in json format
```
a little typo! similarily in other areas; 🏷️

File: docs/_data/mesheryctlcommands/cmds.yml
Comment (alphaX86) [reply docs/_data/mesheryctlcommands/cmds.yml:539]: Is it me that I feel that this PR is dragging me too much or no? 🙂 (Sigh) I'll leave no mistake behind

Comment (afzal442) [reply docs/_data/mesheryctlcommands/cmds.yml:539]: no, it's alr8! it happens as it is OS! #collobaration first, code second! 😄

**Comment by l5io** [comment]
🚀 Preview for commit 74c716306a4652a2e86d9907b850671325bea75d at: https://614bedcc11d1a2134d6e116d--meshery-docs-preview.netlify.app

**Review by afzal442 — APPROVED** [review]
LGTM! Thanks! AlphaPro.

**Review by navendu-pottekkat — APPROVED** [review]
LGTM

**Pull request merged**
